### PR TITLE
(ssis-vs2019) Add package for Visual Studio 2019 SSIS extension

### PR DESCRIPTION
**Pull request by Alea81**

### CODE REVIEW

File: automatic/sql-server-data-tools/legal/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,164 @@
+GNU LESSER GENERAL PUBLIC LICENSE
+Version 3, 29 June 2007
+
+Copyright (C) 2016-2017 SonarSource SA
```
Comment (pascalberger) [inline automatic/sql-server-data-tools/legal/LICENSE.txt]: This seems to be the wrong license

File: automatic/sql-server-data-tools/legal/VERIFICATION.TXT [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+1. Download <https://ssis.gallerycdn.vsassets.io/extensions/ssis/sqlserverintegrationservicesprojects/3.0/1555036193942/Microsoft.DataTools.IntegrationServices.exe>
+2. Then use one of the following methods to obtain the checksum
+  - Use powershell function 'Get-Filehash'
+  - Use chocolatey utility 'checksum.exe'
+
+  checksum type: sha256
+  checksum: F96D1152D442A9FFA46C02D71DBEAED4AFC09A9A4B25564D4AB927FD1BE1D38C
+
+File 'LICENSE.txt' is obtained from <https://marketplace.visualstudio.com/items/SonarSource.SonarLintforVisualStudio/license>
```
Comment (pascalberger) [inline automatic/sql-server-data-tools/legal/VERIFICATION.TXT]: Wrong link to obtain the license from

File: automatic/sql-server-data-tools/sql-server-data-tools.nuspec [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>sql-server-data-tools</id>
```
Comment (pascalberger) [inline automatic/sql-server-data-tools/sql-server-data-tools.nuspec]: This is not SQL Server Data Tools (which was the name of the whole installer) but only the extension for integration services projects. I would call it `ssis-vs2019`.

File: automatic/sql-server-data-tools/sql-server-data-tools.nuspec [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <title>Microsoft SQL Server Integration Services Projects</title>
+    <version>3.0</version>
+    <authors>Microsoft</authors>
+    <owners>BBT Software AG</owners>
+    <projectUrl>https://marketplace.visualstudio.com/items?itemName=SSIS.SqlServerIntegrationServicesProjects</projectUrl>
+    <packageSourceUrl>https://github.com/bbtsoftware/chocolatey-packages/tree/master/automatic/sql-server-data-tools</packageSourceUrl>
+    <!-- todo: icon -->
+    <iconUrl>https://cdn.jsdelivr.net/gh/bbtsoftware/chocolatey-packages@eca9d801fec67aa11df1af98be470d9ab601c5cf/icons/dotnet.png</iconUrl>
```
Comment (pascalberger) [inline automatic/sql-server-data-tools/sql-server-data-tools.nuspec]: Wrong icon (icon file is also missing). There's also a hash in the URL, therefore URL would still be wrong if file is commited. But I think we have a script in place to update this.

File: automatic/sql-server-data-tools/sql-server-data-tools.nuspec [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <summary>This project may be used for building high performance data integration and workflow solutions, including extraction, transformation, and loading (ETL) operations for data warehousing.</summary>
+    <description>
+This project may be used for building high performance data integration and workflow solutions, including extraction, transformation, and loading (ETL) operations for data warehousing.
+
+Visit https://techcommunity.microsoft.com/t5/SQL-Server-Integration-Services/bg-p/SSIS for the latest information, tips, news, and announcements about SSIS directly from the product team.
+    </description>
+    <tags>microsoft .net integration services ssis ssdt sql server data tools</tags>
```
Comment (pascalberger) [inline automatic/sql-server-data-tools/sql-server-data-tools.nuspec]: `sql server` should be `sql-server`

Comment (pascalberger) [inline automatic/sql-server-data-tools/sql-server-data-tools.nuspec]: `data tools` should be `data-tools`

File: automatic/sql-server-data-tools/legal/VERIFICATION.TXT (hunk 2 above)
Comment (Alea81) [reply automatic/sql-server-data-tools/legal/VERIFICATION.TXT]: Have removed the licence file and added a link to the verification.txt

**Comment by Alea81** [comment]
@pascalberger Please review again

File: automatic/ssis-vs2019/tools/ChocolateyInstall.ps1 [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+$toolsPath = Split-Path $MyInvocation.MyCommand.Definition
+$filePath = "$toolsPath\Microsoft.DataTools.IntegrationServices.exe"
+
+$packageArgs = @{
+    packageName   = $env:ChocolateyPackageName
+    softwareName  = 'Microsoft.DataTools.IntegrationServices*'
+    file          = $filePath
+    fileType      = 'exe'
+    silentArgs    = "/quiet /log `"$env:TEMP\$env:ChocolateyPackageName.$env:ChocolateyPackageVersion.log`""
```
Comment (x-jokay) [inline automatic/ssis-vs2019/tools/ChocolateyInstall.ps1]: Should ` and " be invertert before **`"** `$env:TEMP\`

Comment (x-jokay) [inline automatic/ssis-vs2019/tools/ChocolateyInstall.ps1]: Chars right before `$env:TEMP\$env:ChocolateyPackageName` may be inverted?

File: automatic/ssis-vs2019/sql-server-data-tools.nuspec [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    <packageSourceUrl>https://github.com/bbtsoftware/chocolatey-packages/tree/master/automatic/ssis-vs2019</packageSourceUrl>
+    <iconUrl>https://cdn.jsdelivr.net/gh/bbtsoftware/chocolatey-packages@ddc4c769721f219a370efae9492bb333dce52627/icons/ssis-vs2019.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <summary>This project may be used for building high performance data integration and workflow solutions, including extraction, transformation, and loading (ETL) operations for data warehousing.</summary>
+    <description>
+This project may be used for building high performance data integration and workflow solutions, including extraction, transformation, and loading (ETL) operations for data warehousing.
+
+Visit https://techcommunity.microsoft.com/t5/SQL-Server-Integration-Services/bg-p/SSIS for the latest information, tips, news, and announcements about SSIS directly from the product team.
+    </description>
+    <tags>microsoft integration-services ssis ssis-vs2019 ssdt sql-server data-tools</tags>
```
Comment (x-jokay) [inline automatic/ssis-vs2019/sql-server-data-tools.nuspec]: You may add `sql` only as tag?

**Comment by Alea81** [comment]
@pascalberger @x-jokay Ready to review

File: automatic/ssis-vs2019/legal/VERIFICATION.TXT [hunk 8]
```diff
@@ -0,0 +1,16 @@
+VERIFICATION
```
Comment (pascalberger) [inline automatic/ssis-vs2019/legal/VERIFICATION.TXT]: VERIFICATION file (and whole legal folder) is not required if extension is not embedded and shout be removed

File: automatic/ssis-vs2019/ssis-vs2019.nuspec [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    <description>
+This project may be used for building high performance data integration and workflow solutions, including extraction, transformation, and loading (ETL) operations for data warehousing.
+
+Visit https://techcommunity.microsoft.com/t5/SQL-Server-Integration-Services/bg-p/SSIS for the latest information, tips, news, and announcements about SSIS directly from the product team.
+    </description>
+    <tags>microsoft integration integration-services sql ssis ssis-vs2019 ssdt sql-server data-tools</tags>
+    <dependencies />
+  </metadata>
+  <files>
+    <file src="legal\**" target="legal" />
```
Comment (pascalberger) [inline automatic/ssis-vs2019/ssis-vs2019.nuspec]: If `legal` folder is removed this is no longer required.

File: automatic/ssis-vs2019/update1.ps1 [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import-module au
+
+$releases = 'https://marketplace.visualstudio.com/items?itemName=SSIS.SqlServerIntegrationServicesProjects'
+
+function global:au_SearchReplace {
+  @{
```
Comment (pascalberger) [inline automatic/ssis-vs2019/update1.ps1]: Can be removed if VERIFICATION file no longer exists

**Comment by Alea81** [comment]
@pascalberger @x-jokay Ready to review

**Comment by Alea81** [comment]
do we need a short readme here?

**Comment by pascalberger** [comment]
> @Alea81 wrote:
> do we need a short readme here?

Yes, a readme like [here](https://github.com/bbtsoftware/chocolatey-packages/tree/master/automatic/markdownlint-cli) would be good. The content of the readme (except the first two lines) will be used for the description in the `.nuspec` file.

**Comment by Alea81** [comment]
@pascalberger readme there, ready for review

File: automatic/ssis-vs2019/tools/ChocolateyInstall.ps1 [hunk 11]
```diff
@@ -0,0 +1,14 @@
+﻿$ErrorActionPreference = 'Stop';
+
+$packageArgs = @{
+    packageName   = $env:ChocolateyPackageName
+    softwareName  = 'Microsoft.DataTools.IntegrationServices*'
+    fileType      = 'exe'
+    silentArgs    = "/quiet /log `"$env:TEMP\$env:ChocolateyPackageName.$env:ChocolateyPackageVersion.log`""
+    validExitCodes= @(0)
+    url           = "https://ssis.gallerycdn.vsassets.io/extensions/ssis/sqlserverintegrationservicesprojects/3.0/1555036193942/Microsoft.DataTools.IntegrationServices.exe"
+    checksum      = "f96d1152d442a9ffa46c02d71dbeaed4afc09a9a4b25564d4ab927fd1be1d38c"
```
Comment (pascalberger) [inline automatic/ssis-vs2019/tools/ChocolateyInstall.ps1]: `url` and `checksum` don't need to be set necessarily as they will be set by the update script after a new version is detected if version in `.nuspec` is not set

Comment (Alea81) [reply automatic/ssis-vs2019/tools/ChocolateyInstall.ps1]: Have changed it to "update". Empty string is not recognized... Will be updated with the first release.

**Comment by Alea81** [comment]
@pascalberger readme there, ready for review

File: automatic/ssis-vs2019/tools/ChocolateyInstall.ps1 [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+    softwareName  = 'Microsoft.DataTools.IntegrationServices*'
+    fileType      = 'exe'
+    silentArgs    = "/quiet /log `"$env:TEMP\$env:ChocolateyPackageName.$env:ChocolateyPackageVersion.log`""
+    validExitCodes= @(0)
+    url           = "update"
+    checksum      = "update"
+    checksumType  = "sha256"
+  }
+  
+Install-ChocolateyPackage @packageArgs
```
Comment (pascalberger) [inline automatic/ssis-vs2019/tools/ChocolateyInstall.ps1]: We might want to check if `devenv.exe` is running before trying to install to give a better error message. I created #56 for tracking this.

**Pull request merged**